src/messaging/tests/TestHelpers.h

```diff
@@ -0,0 +1,71 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <messaging/ExchangeContext.h>
+#include <messaging/ExchangeMgr.h>
+#include <transport/SecureSessionMgr.h>
+#include <transport/TransportMgr.h>
+#include <transport/raw/tests/NetworkTestHelpers.h>
+
+namespace chip {
+namespace Test {
+
+class MessagingContext : public IOContext
+{
+public:
+    MessagingContext() :
+        mPeer(Transport::PeerAddress::UDP(GetAddress(), CHIP_PORT)),
+        mPairingPeerToLocal(Optional<NodeId>::Value(GetSourceNodeId()), GetLocalKeyId(), GetPeerKeyId()),
+        mPairingLocalToPeer(Optional<NodeId>::Value(GetDestinationNodeId()), GetPeerKeyId(), GetLocalKeyId())
+    {}
+
+    /// Initialize the underlying layers and test suite pointer
+    CHIP_ERROR Init(nlTestSuite * suite, TransportMgrBase * transport);
+
+    // Shutdown all layers, finalize operations
+    CHIP_ERROR Shutdown();
+
+    static Inet::IPAddress GetAddress()
+    {
+        Inet::IPAddress addr;
+        Inet::IPAddress::FromString("127.0.0.1", addr);
+        return addr;
+    }
+    static constexpr NodeId GetSourceNodeId() { return 123654; }
```

Comment:
why are these methods instead of `static constexpr kSourceNodeId = 123654` ?

Reply:
`static constexpr` can not be inline, it require a definition in the .cpp file. It turn out that constexpr inline function is a bit more feasible because functions can be inline, so it doesn't requires an extra defination.

Reply:
Could you double-check this? I believe you can define constexpr in the headers as long as they are static, without any CPP change required.

Reply:
I hve found the problem. Following code is using `Optional<NodeId>::Value`, which takes a reference, forcing the field to be allocated.

```
    MessagingContext() :
        mPeer(Transport::PeerAddress::UDP(GetAddress(), CHIP_PORT)),
        mPairingPeerToLocal(Optional<NodeId>::Value(GetSourceNodeId()), GetLocalKeyId(), GetPeerKeyId()),
        mPairingLocalToPeer(Optional<NodeId>::Value(GetDestinationNodeId()), GetPeerKeyId(), GetLocalKeyId())
    {}
```


C++17 will fix the issue. Anyway, using the inline constexpr function will return a temporary object, which shouldn't causing any problem.